avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/FeatureIndicesConversion.scala

```diff
@@ -140,39 +158,48 @@ object FeatureIndicesConversion {
    *
    * @param ntvs A seq of NTV in Row type
    * @param featureMapping The mapping of name+term to id
+   * @param discardUnknownEntries Whether to discard unknown entries
    * @return A seq of IdValue
    */
-  private def convertNTVToIdValues(ntvs: Seq[Row], featureMapping: Map[String, Long]): Seq[TensorizeIn.IdValue] = {
+  private def convertNTVToIdValues(
+    ntvs: Seq[Row],
+    featureMapping: Map[String, Long],
+    discardUnknownEntries: Boolean): Seq[TensorizeIn.IdValue] = {
 
     // The number of unique name-term combinations
     val cardinality = featureMapping.size.toLong
+    val lastIndex = if (discardUnknownEntries) cardinality - 1 else cardinality
```

Comment:
please explain this line. Does cardinality include "unknown" or not?
if cardinality counts "unknown" and discardUnknownEntries is true, then lastIndex = cardinality-2.
if cardinality does not count "unknown" and discardUnknownEntries is true, then lastIndex = cardinality-1.
right?

Reply:
`cardinality` never includes the unknown. so yes, it's cardinality-1 when discarding the unknowns